Tensile/KernelWriterAssembly.py

```diff
@@ -8790,13 +8802,16 @@ def localWriteDo(self, kernel, tP, uDu=0):
               if self.db["ForceInputValue%s"%tc]:
                 localWriteCode.addInst("v_mov_b32", vgpr("G2L%s+%u"%(tc, g2lIdx)), self.db["ForceValue%s"%tc], "ForceInputValue")
               if kernel["ProblemType"]["Fp16AltImpl"]:
-                numIters = 1 if blockWidth <= 1 else blockWidth 
+                numIters = 1 if blockWidth <= 1 else blockWidth
                 for iter in range(0, numIters):
-                   vgprsrc = vgpr("G2L%s+%u"%(tc, g2lIdx+iter))
-                   vgprsrc += " src0_sel:WORD_1"
-                   localWriteCode.addInst("v_cvt_f32_f16", vgpr("G2Lpipe0"), vgpr("G2L%s+%u"%(tc, g2lIdx+iter)),"")
-                   localWriteCode.addInst("v_cvt_f32_f16", vgpr("G2Lpipe1"), vgprsrc,"")
-                   localWriteCode.addInst("v_pack_b32_f16", vgpr("G2L%s+%u"%(tc, g2lIdx+iter)), vgpr("G2Lpipe0"),vgpr("G2Lpipe1"), "op_sel:[1,1,0]","")
+                  vgprsrc = vgpr("G2L%s+%u"%(tc, g2lIdx+iter))
+                  localWriteCode.addInst("v_cvt_f32_f16", vgpr("G2Lpipe0"), vgprsrc,"")
+                  localWriteCode.addInst("v_cvt_f32_f16", vgpr("G2Lpipe1"), vgprsrc, "src0_sel:WORD_1", "")
+
+                  localWriteCode.addInst("v_add_u32", vgpr("G2Lpipe0"), sgpr("Fp16AltRound"), vgpr("G2Lpipe0"), "")
```

Comment:
I am not sure how v_cvt_f32_f16 works if the value is NaN.
Is this safer if we check the value is NaN or not and add rounding only when it is not NaN?

Reply:
I was debating putting that in, as that would add even more extra instructions on the existing functionality which might cause a performance regression. I'll add it though to be safe.